File: java/src/sdk/dataobjects/paypal/payflow/CheckPayment.java
package paypal.payflow;

/**
 * Used for Check Payment related information.
 * CheckPayment is associated with CheckTender. {@link CheckTender}
 */
public final class CheckPayment extends PaymentDevice {

    /**
     * Constructor
     *
     * @param micr MICR Value
     *             This is used as Payment Device for the CheckTender.
     *              * <p>
     *  Maps to Payflow Parameter: MICR
     * //Create the CheckPayment object
     * CheckPayment payDevice = new CheckPayment("XXXXXXXXXX");
     * </p>
     */
    public CheckPayment(String micr) {
        super(micr);
    }

    protected void generateRequest() {
//    	Put the base field Acct as MICR.
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MICR, super.getAcct()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_NAME, super.getName()));
    }

}

File: dotNET/PayflowProSDK/PayPal/Payments/DataObjects/CheckPayment.cs
#region "Imports"

using System;
using PayPal.Payments.Common.Utility;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for Check Payment related information
	/// </summary>
	/// <remarks>
	/// CheckPayment is associated with CheckTender.
	/// <seealso cref="CheckTender"/>
	/// </remarks>
	public sealed class CheckPayment : PaymentDevice
	{
		#region "Constructors"

		/// <summary>
		/// Constructor.
		/// </summary>
		/// <remarks>This is used as Payment Device for the CheckTender.
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>Micr --> MICR</code>
		/// </remarks>
		///  <example>
		///  <code lang="C#" escaped="false">
		///		.............
		///		
		///		CheckPayment PayDevice = new CheckPayment("XXXXXXXXXX");
		///		
		///		..............
		///  </code>
		///  <code lang="C#" escaped="false">
		///		.............
		///		
		///		Dim PayDevice As CheckPayment = new CheckPayment("XXXXXXXXXX")
		///		
		///		..............
		///  </code>
		///  </example>
		///  <seealso cref="CheckTender"/>
		/// <param name="Micr">MICR value</param>
		public CheckPayment(String Micr) : base(Micr)
		{
		}

		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal override void GenerateRequest()
		{
			try
			{
				//Put the base field Acct as MICR.
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MICR, base.Acct));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_NAME, base.Name));
			}
			catch (BaseException)
			{
				throw;
			}
			catch (Exception Ex)
			{
				DataObjectException DEx = new DataObjectException(Ex);
				throw DEx;
			}
            //catch
            //{
            //    throw new Exception();				
            //}
		}

		#endregion
	}

}